### ISSUE TITLE
app:config:import fails with "Please specify the admin custom URL."

### ISSUE THREAD

**Issue by BlitzInternet** [magento/magento2#15090]
When trying to import a dumped config on another server (demo), the import fails. Nothing in the config file has been changed but URL and password values.
The (redacted) dump: [env.redacted.txt](https://github.com/magento/magento2/files/1986815/env.redacted.txt)
Problem is caused in L106 - L111

I've tried adding the 'use_custom' => 1, 'use_custom_path' => 1, but it didn't help. Setting both of those to 0 does fix the issue, but as expected then the custom admin url/path isn't used.

In magento/module-config/Model/Config/Backend/Admin/Usecustom.php where the error is thrown, debugging reveals that no other config keys are set in $this->_data. As such, getData('groups/url/fields/custom/value') will always return null.

### Preconditions
1. mage 2.2.4
2. ubuntu 16.04
3. php 7.0.28

### Steps to reproduce
1. export config on dev machine containing custom admin url and path
2. adjust urls for demo server without adding, modifying or deleting any keys
3. import config on demo server

### Expected result
1. sucessful import

### Actual result
> php bin/magento app:config:import
Processing configurations data from configuration file...
Import failed: Please specify the admin custom URL.

**Comment by avaldi** [comment]
Same issue here. @BlitzInternet did you manage to find a workaround?

**Comment by BlitzInternet** [comment]
I disabled the check via plugin. Nothing else you can do.

**Comment by engcom-backlog-nickolas** [comment]
Hello @BlitzInternet, thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by davidalger** [comment]
Run into this over here as well. Used this patch to workaround the issue for now (I may submit a PR if I can find the time this week but want to share here in case I don't):

```
diff --git a/vendor/magento/module-config/Model/Config/Backend/Admin/Usecustom.php b/vendor/magento/module-config/Model/Config/Backend/Admin/Usecustom.php
--- a/vendor/magento/module-config/Model/Config/Backend/Admin/Usecustom.php	2019-04-08 11:24:09.000000000 -0500
+++ b/vendor/magento/module-config/Model/Config/Backend/Admin/Usecustom.php	2019-04-08 11:24:03.000000000 -0500
@@ -56,8 +56,9 @@
     {
         $value = $this->getValue();
         if ($value == 1) {
-            $customUrl = $this->getData('groups/url/fields/custom/value');
-            if (empty($customUrl)) {
+            $customUrlField = $this->getData('groups/url/fields/custom/value');
+            $customUrlConfig = $this->_config->getValue('admin/url/custom');
+            if (empty($customUrlField) && empty($customUrlConfig)) {
                 throw new \Magento\Framework\Exception\LocalizedException(__('Please specify the admin custom URL.'));
             }
         }
```

I'm applying this to the site we ran into it on using this atm: https://github.com/cweagans/composer-patches

**Comment by magento-engcom-team** [comment]
Hi @BlitzInternet. Thank you for your report. 
The issue has been fixed in magento/magento2#22281 by @davidalger in 2.3-develop branch
Related commit(s):
 - [8434d74abd6393d2cb4ad45eff6c9ec292f4dc71](https://github.com/magento/magento2/commit/8434d74abd6393d2cb4ad45eff6c9ec292f4dc71)
 - [e340978f1ef25bb39d5e03939ddad30b0c927eab](https://github.com/magento/magento2/commit/e340978f1ef25bb39d5e03939ddad30b0c927eab)

The fix will be available with the upcoming 2.3.2 release.

**Comment by magento-engcom-team** [comment]
Hi @BlitzInternet. Thank you for your report. 
The issue has been fixed in magento/magento2#22282 by @davidalger in 2.2-develop branch
Related commit(s):
 - [e12eeaac9a1892bab43b8ab7083e03922da3b2ab](https://github.com/magento/magento2/commit/e12eeaac9a1892bab43b8ab7083e03922da3b2ab)

The fix will be available with the upcoming 2.2.9 release.
